src/controllers/confirm.company.controller.ts

```diff
@@ -21,23 +22,29 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = async (req: Request, res: Response) => {
+export const post = async (req: Request, res: Response, next: NextFunction) => {
   const session: Session = req.session as Session;
   const companyNumber = req.query["companyNumber"] as string;
-  const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, companyNumber);
-  if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
-    await createNewConfirmationStatement(req);
-    const url = TRADING_STATUS_PATH.replace(":companyNumber", companyNumber);
-    return res.redirect(url);
-  } else {
-    return displayEligibilityStopPage(res, eligibilityStatusCode );
+  try {
+    const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, companyNumber);
+    if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
+      await createNewConfirmationStatement(req);
+      const url = TRADING_STATUS_PATH.replace(":companyNumber", companyNumber);
+      return res.redirect(url);
+    } else {
+      return displayEligibilityStopPage(res, eligibilityStatusCode);
+    }
+  } catch (e) {
+    logger.errorRequest(req, `An error has occurred. Re-routing to the error screen - ${e.stack}`);
```

Comment:
line 38 looks exactly the same as the log message in the error controller, we don't want to log the same thing twice. I'd delete line 38 and the error controller has 
`logger.errorRequest(req, `An error has occurred. Re-routing to the error screen - ${err.stack}`);` already in it so it will log out the message.